packages/commonwealth/client/scripts/views/components/sidebar/index.tsx

```diff
@@ -19,99 +16,100 @@ import { GovernanceSection } from './governance_section';
 import { SidebarQuickSwitcher } from './sidebar_quick_switcher';
 import { CWIcon } from '../component_kit/cw_icons/cw_icon';
 import { CWText } from '../component_kit/cw_text';
-import withRouter from '../../../navigation/helpers';
+import { useCommonNavigate } from '../../../navigation/helpers';
+import useForceRerender from 'hooks/useForceRerender';
 
 export type SidebarMenuName =
   | 'default'
   | 'createContent'
   | 'exploreCommunities';
 
-class SidebarComponent extends ClassComponent {
-  view() {
-    const activeAddressRoles = app.roles.getAllRolesInCommunity({
-      chain: app.activeChainId(),
-    });
+export const Sidebar = () => {
+  const navigate = useCommonNavigate();
+  const forceRerender = useForceRerender();
 
-    app.sidebarRedraw.on('redraw', () => {
-      this.redraw();
-    });
+  const activeAddressRoles = app.roles.getAllRolesInCommunity({
+    chain: app.activeChainId(),
+  });
 
-    const currentChainInfo = app.chain?.meta;
+  useEffect(() => {
+    app.sidebarRedraw.on('redraw', () => forceRerender());
 
-    const onHomeRoute =
-      _DEPRECATED_getRoute() === `/${app.activeChainId()}/feed`;
+    return () => {
+      app.sidebarRedraw.on('redraw', () => forceRerender());
+    };
+  });
 
-    const hideChat =
-      !currentChainInfo ||
-      !activeAddressRoles ||
-      !isActiveAddressPermitted(
-        activeAddressRoles,
-        currentChainInfo,
-        Action.VIEW_CHAT_CHANNELS
-      );
+  const currentChainInfo = app.chain?.meta;
 
-    const isAdmin =
-      app.user.isSiteAdmin ||
-      app.roles.isAdminOfEntity({
-        chain: app.activeChainId(),
-      });
+  const onHomeRoute = _DEPRECATED_getRoute() === `/${app.activeChainId()}/feed`;
```

Comment:
if you refactor the file, do not use _DEPRECATED_getRoute, we should get rid of it eventually

Reply:
I tried to replace with `useLocation` but I wasn't sure how to use it here.

Reply:
`const { path } = useLocation()` should do the job